pubsub/tests/system.py

```diff
@@ -99,35 +100,34 @@ def test_subscribe_to_messages(
     subscriber.create_subscription(subscription_path, topic_path)
     subscription = subscriber.subscribe(subscription_path)
 
+    messages = 50
+
+    data = {
+        str(index): b'Wooooo! The claaaaaw!' for
+        index in six.moves.range(messages)
+    }
+
     # Publish some messages.
     futures = [
         publisher.publish(
             topic_path,
-            b'Wooooo! The claaaaaw!',
+            data[str(index)],
             num=str(index),
         )
-        for index in six.moves.range(50)
+        for index in six.moves.range(messages)
     ]
 
     # Make sure the publish completes.
     for future in futures:
         future.result()
 
-    # Actually open the subscription and hold it open for a few seconds.
     # The callback should process the message numbers to prove
     # that we got everything at least once.
     callback = AckCallback()
     subscription.open(callback)
-    for second in six.moves.range(10):
-        time.sleep(1)
-
-        # The callback should have fired at least fifty times, but it
-        # may take some time.
-        if callback.calls >= 50:
-            return
-
-    # Okay, we took too long; fail out.
-    assert callback.calls >= 50
+    while len(set(callback.data.keys())) < messages:
```

Comment:
Why did you change this? You took a test that had a potential failure case that would fail and made it into a failure case that would loop forever, and for no obvious gain that I can see.

Also, a `while/pass` is not guaranteed to surrender the GIL. (As in, it likely will not.)

Reply:
I thought that it was required to deliver the messages at least once and that "taking too long" shouldn't be a failure case. 

I guess I was wrong.  I'm changing the test back.